frontend/src/components/administration/users/dialog-edit-user.tsx

```diff
@@ -1,148 +1,155 @@
-import { ChangeEvent, FC, useEffect, useState } from 'react';
+import { FC, useEffect } from 'react';
+import { SubmitHandler } from 'react-hook-form';
 
-import { User, UserRole, UserSex } from '@autoline/shared/common/types/types';
+import CrossIcon from '@assets/images/edit-profile/cross.svg';
+import { User } from '@autoline/shared/common/types/types';
+import { editUserSchema } from '@autoline/shared/validation-schemas';
 import { ButtonFill } from '@components/common/button-fill/button-fill';
 import { ButtonOutline } from '@components/common/button-outline/button-outline';
 import { InputField } from '@components/common/input-field/input-field';
-import { SelectField } from '@components/common/select-field/select-field';
-import Container from '@mui/material/Container';
-import Dialog from '@mui/material/Dialog';
-import DialogActions from '@mui/material/DialogActions';
-import DialogContent from '@mui/material/DialogContent';
-import DialogTitle from '@mui/material/DialogTitle';
-import FormControl from '@mui/material/FormControl';
+import { SelectFieldForm } from '@components/edit-profile/select-field-form/select-field-form';
+import { useAppForm } from '@hooks/hooks';
 import MenuItem from '@mui/material/MenuItem';
-import { SelectChangeEvent } from '@mui/material/Select';
+import Modal from '@mui/material/Modal';
 
-type DialogEditUserProps = {
-  user?: User;
-  handleClose: () => void;
-  handleSubmit: (user: User, newUserData: Partial<Omit<User, 'id'>>) => void;
-};
-
-const DialogEditUser: FC<DialogEditUserProps> = (props) => {
-  const { user, handleClose, handleSubmit } = props;
-  const [role, setRole] = useState<UserRole | undefined>(user?.role);
-  const [name, setName] = useState<string | undefined>(user?.name);
-  const [email, setEmail] = useState<string | undefined>(user?.email);
-  const [phone, setPhone] = useState<string | undefined>(user?.phone);
-  const [location, setLocation] = useState<string | undefined>(user?.location);
-  const [sex, setSex] = useState<UserSex | undefined>(user?.sex);
+import styles from './styles.module.scss';
 
-  useEffect(() => {
-    if (!user) return;
-    setName(user.name);
-    setEmail(user.email);
-    setPhone(user.phone);
-    setLocation(user.location);
-    setSex(user.sex);
-    setRole(user.role);
-  }, [user]);
+interface DialogEditUserProps {
+  user?: User;
+  onClose: () => void;
+  onSave: (user: User, newUserData: Partial<Omit<User, 'id'>>) => void;
```

Comment:
Usually, `onClosed` passes as the last prop

Reply:
Done